src/booking/Offers/OfferTypes.ts

```diff
@@ -550,6 +571,81 @@ export interface OfferSliceSegmentPassenger {
    * fare basis code is not available or the airline does not use fare basis codes.
    */
   fare_basis_code: string
+
+  /**
+   * Data about the cabin that the passenger will be flying in for this segment
+   */
+  cabin: {
+    /**
+     * The name of the cabin class
+     */
+    name: CabinClass
+
+    /**
+     * TThe name that the marketing carrier uses to market this cabin class
+     */
+    marketing_name: string
+
+    /**
+     * The amenities specific to this cabin class on this plane
+     */
+    amenities: {
+      /**
+       * If Wi-Fi is available, information on its cost, availability, etc
+       */
+      wifi: {
+        /**
+         * Whether Wi-Fi is available in this cabin
+         */
+        available: boolean
+
+        /**
+         * The cost, if any, to use the Wi-Fi
+         */
+        cost: 'free' | 'paid' | 'free or paid' | 'n/a'
+      } | null
```

Comment:
Would it be helpful to isolate these definitions into their own types and reference them in here? Not a must of course, but if you ever want to reference the wifiAmmenity type without the nullable (i.e. after doing the null check) it won't have it immediately available, it'd probably require some utility class wrapper. 

 ```suggestion
        wifi: WifiAmmenity | null
```